src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -1001,3 +1003,73 @@ void TieredHNSWIndex<DataType, DistType>::TieredHNSW_BatchIterator::filter_irrel
     // Update number of results (pop the tail)
     array_pop_back_n(rl.results, end - cur_end);
 }
+
+template <typename DataType, typename DistType>
+VecSimIndexInfo TieredHNSWIndex<DataType, DistType>::info() const {
+    auto info = VecSimTieredIndex<DataType, DistType>::info();
```

Comment:
Can you explain why you removed the general purpose tiered info implementation?
I think you could have called it and just inject the two lines
```
    HnswTieredInfo hnswTieredInfo = {.pendingSwapJobsThreshold = this->pendingSwapJobsThreshold};
    info.tieredInfo.specificTieredBackendInfo.hnswTieredInfo = hnswTieredInfo;
```

Reply:
I thought it was better and cleaner like this (that the derived class sets its own info), but not critical, I can move the backend data info back to the base class.